From 4390c8c8ddd9eaa0a20432c9a0d00fade6afab89 Mon Sep 17 00:00:00 2001
From: Lyubo <lyubodzhamov@gmail.com>
Date: Fri, 12 Apr 2019 17:51:17 -0400
Subject: [PATCH] Status bar swipe disalow for widgets sheet

* Fixes: https://gitlab.com/LineageOS/issues/android/issues/372.

Status bar swipe feature interferes with scrolling when enough widgets to scroll through.
If not enough widgets to scroll through, the swipe down will close the widgets sheet, so there is no use for this feature here.

Change-Id: Icecf1f0f8d093a2f5c61d1a0ce7975cbc98cf2a2
---
 src/com/android/launcher3/widget/WidgetsFullSheet.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/com/android/launcher3/widget/WidgetsFullSheet.java b/src/com/android/launcher3/widget/WidgetsFullSheet.java
index e94d81d75..489d5860c 100644
--- a/src/com/android/launcher3/widget/WidgetsFullSheet.java
+++ b/src/com/android/launcher3/widget/WidgetsFullSheet.java
@@ -196,7 +196,8 @@ public class WidgetsFullSheet extends BaseWidgetSheet
 
     @Override
     protected boolean isOfType(int type) {
-        return (type & TYPE_WIDGETS_FULL_SHEET) != 0;
+        return (type & TYPE_WIDGETS_FULL_SHEET) != 0 ||
+                (type & TYPE_STATUS_BAR_SWIPE_DOWN_DISALLOW) != 0;
     }
 
     @Override
-- 
2.17.1

